client/components/Item/MediaStreamSelector.vue

```diff
@@ -0,0 +1,210 @@
+    /**
+     * Media streams to display in the selector
+     */
+    mediaStreams: {
+      type: Array as PropType<MediaStream[]>,
+      default: () => []
+    },
+    /**
+     * Overrides the default track number
+     */
+    defaultStreamIndex: {
+      type: Number,
+      default: undefined
+    }
+  },
+  data() {
+    return { trackIndex: -1 as number };
+  },
+  computed: {
+    type: {
+      /**
+       * Calculates the type for the given media streams
+       *
+       * @returns {string|undefined} Type of the given media streams
+       */
+      get(): string | undefined {
+        return this.mediaStreams[0].Type;
+      }
+    },
+    selectItems: {
+      /**
+       * Used to model the media stream index as a value and the potential strings
+       *
+       * @returns {{text: SelectItems, value: number}[]} List of objects prepared for Vuetify v-select with the strings to display as "text" and index number as "value".
+       */
+      get(): { text: SelectItems; value: number | undefined }[] {
+        const items = this.mediaStreams.map((value, _idx) => {
+          return {
+            text: {
+              selection: this.getTrackSelection(value),
+              subtitle: this.getTrackSubtitle(value),
+              icon: this.getTrackIcon(value),
+              title: this.getTrackTitle(value)
+            },
+            value: value.Index
+          };
+        });
+
+        items.unshift({
+          value: -1,
+          text: {
+            selection: this.$t('disabled'),
+            title: this.$t('disabled'),
+            subtitle: undefined,
+            icon: undefined
+          }
+        });
+
+        return items;
+      }
+    },
+    /**
+     * @returns {number|undefined} Default index to use (undefined if none)
+     */
+    defaultIndex: {
+      get(): number | undefined {
+        if (this.defaultStreamIndex !== undefined) {
+          return this.defaultStreamIndex;
+        }
+
+        return this.mediaStreams.find((track) => track.IsDefault)?.Index;
+      }
+    }
+  },
+  /**
+   * Sets the default track when loading the component
+   */
+  beforeMount() {
+    if (this.defaultIndex !== undefined) {
+      this.trackIndex = this.defaultIndex;
+    }
+
+    this.$emit('input', this.trackIndex);
+  },
+  methods: {
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string} Text to display in select when track is choosen
+     */
+    getTrackSelection(track: MediaStream): string {
+      if (track.DisplayTitle) {
+        return track.DisplayTitle;
+      }
+
+      return '';
+    },
+    /**
+     * @param {MediaStream} track - Track to parse
+     * @returns {string|undefined} Optional icon to use for the track line in the v-select menu
+     */
+    getTrackIcon(track: MediaStream): string | undefined {
+      if (this.type === 'Audio' && track.ChannelLayout) {
+        return this.getSurroundIcon(track.ChannelLayout);
+      }
+
+      return undefined;
```

Comment:
This is unnecessary, as not returning anything will give you `undefined` anyway.